runners/direct-java/src/test/java/org/apache/beam/runners/direct/DirectMetricsTest.java

```diff
@@ -125,6 +129,55 @@ public void testApplyAttemptedCountersQueryOneNamespace() {
             committedMetricsResult("ns1", "name1", "step2", 0L)));
   }
 
+  private boolean matchesScopeWithSingleFilter(String actualScope, String filter) {
```

Comment:
May be easier to just make the sub-path matching method public and test that directly?